paragraph/graph/graph.cc

```diff
@@ -74,6 +74,13 @@ bool Graph::IsConnected() {
   }
 }
 
+absl::Status Graph::TrimGraph() {
```

Comment:
Name is probably misleading. It actually trims all empty instructions and subroutines recursively, but only for a subset of "trivial" instructions. If it's a general purpose or call instructions, we check if inside inner subroutines all instructions don't have seconds as payload, and then we delete whole subroutine. If instruction afterwards doesn't have inner subroutines and seconds associated with it, it can be dropped with its parent subroutine further. But we don't drop subroutines and instructions in these subroutines for While, Conditional, and all communication instructions.
We need a function like this to parse logs that have time assigned to fused instructions. Without graph trimming, simulators get only their leaf instructions, which times are not assigned in the trace.